gradio/components/image.py

```diff
@@ -101,6 +102,7 @@ def __init__(
             key: if assigned, will be used to assume identity across a re-render. Components that have the same key across a re-render will have their value preserved.
             mirror_webcam: If True webcam will be mirrored. Default is True.
             show_share_button: If True, will show a share icon in the corner of the component that allows user to share outputs to Hugging Face Spaces Discussions. If False, icon does not appear. If set to None (default behavior), then the icon appears if this Gradio app is launched on Spaces, but not otherwise.
+            stream_frequency: The number of times per second to stream the webcam feed. Only valid if streaming is True.
```

Comment:
Is it true that this argument is valid only if streaming is True? In your example demo, you have this line:

>             input_img = gr.Image(sources=["webcam"], stream_frequency=10, type="numpy")

Reply:
I'm a little confused about the `streaming` argument actually. Is it used at all for input streaming? If I understand correctly, it is only used to set the behavior when a component is being used as output. We should make this clearer in the docs and docstrings.

Reply:
Yes `streaming` parameter is only used for outputs. Will make a note of that in docstring.

Reply:
We do have this logic:

```
            self.sources = (
                ["webcam"] if streaming else ["upload", "webcam", "clipboard"]
            )
```

but I think we can change this to: if an image component only has .stream() events, then the default source is webcam

Reply:
Perhaps we should rename the `streaming` parameter to `stream_out` to help users know which flags they need to set for which use case (if we are worried about breaking, then we could at least deprecate the former and assign its value to the latter)